ui/lib/pki/addon/routes/overview.js

```diff
@@ -7,28 +7,15 @@ export default class PkiOverviewRoute extends Route {
   @service auth;
   @service store;
 
-  get win() {
-    return this.window || window;
-  }
-
   hasConfig() {
     // When the engine is configured, it creates a default issuer.
     // If the issuers list is empty, we know it hasn't been configured
-    const endpoint = `${this.win.origin}/v1/${this.secretMountPath.currentPath}/issuers?list=true`;
-
-    return this.auth
-      .ajax(endpoint, 'GET', {})
+    return this.store
+      .query('pki/issuer', { backend: this.secretMountPath.currentPath })
       .then(() => true)
       .catch(() => false);
   }
 
-  async fetchEngine() {
-    const model = await this.store.query('secret-engine', {
-      path: this.secretMountPath.currentPath,
-    });
-    return model.get('firstObject');
-  }
-
```

Comment:
The `secret-engine` model is passed to the engine from the parent route and we can access it with `this.modelFor('application')` rather than making another network request.

Reply:
Again, any chance by way of navigation you wouldn't hit that parent route? I suspect not, but just coming back to the project I thought I should ask.

Reply:
Good question. Since this engine is mounted as a child under `vault.cluster.secrets.backend` the backend route model hook will fire before we get to the model hooks in the engine so we should always have access to the parent model.